### ISSUE TITLE
Correct resampling logic

### ISSUE THREAD

**Issue by guidocioni** [guidocioni/point_wx#63]
In many functions of `openmeteo_api` a `resample` operation is used to reduce the dataframe size.
Usually the `resample` is combined with the `first()` function to only select the first data point. This works for many variables that have istatanoues values like temperature but does not work for variables that are accumulated on the forecast time step like sunshine duration and total precipitation.
We should fix this operation to add explicit aggregation functions that consider the sum over the resample interval. However it is not straightforward because we have to specify it for every column individually, similarly to what is done in `compute_daily_stats`. 

```python
df.resample('1D', on='time').agg(t2m_mean=('temperature_2m', 'mean'),
                                             t2m_min=('temperature_2m', 'min'),
                                             t2m_max=('temperature_2m', 'max'),
                                             wind_speed_max=(
                                                 'windspeed_10m', 'max'),  # in km/h
                                             daily_rain=(
                                                 'precipitation', 'sum'),  # in mm
                                             daily_snow=(
                                                 'snowfall', 'sum'),  # in cm
                                             cloudcover_mean=(
                                                 'cloudcover', 'mean')  # in %
                                             )
```

But I'm sure we can find a solution

**Comment by guidocioni** [comment]
To solve this it is sufficient to do a rolling before, i.e.

```python
test.rolling(window='3H', on='time').sum().resample('3H', on='time', origin=test.iloc[0]['time']).first()
```

This way, we first do a sum on the preceding 3 hours with `rolling`, and then subset to only get the sum every 3 hours.

**Comment by guidocioni** [comment]
It's hard to completely fix this, but it should be improved in e012bbb11d9f7f95b1a7e88073120fccaab0a805.
